src/ui/camera.js

```diff
@@ -609,6 +617,122 @@ class Camera extends Evented {
         return options;
     }
 
+    _minimumAABBFrustumDistance(tr: Transform, aabb: Aabb): number {
+        const aabbW = aabb.max[0] - aabb.min[0];
+        const aabbH = aabb.max[1] - aabb.min[1];
+        const aabbAspectRatio = aabbW / aabbH;
+        const selectXAxis = aabbAspectRatio > tr.aspect;
+
+        const min = aabb.min;
+        const max = [
+            selectXAxis ? aabb.max[0] : aabb.min[0],
+            selectXAxis ? aabb.min[1] : aabb.max[1],
+            aabb.min[2]
+        ];
+        const axisLength = vec3.length(vec3.sub([], min, max));
+        const minimumDistance = selectXAxis ?
+            axisLength / (2 * Math.tan(tr.fovX * 0.5) * tr.aspect) :
+            axisLength / (2 * Math.tan(tr.fovY * 0.5) * tr.aspect);
+
+        return minimumDistance;
+    }
+
+    _cameraForBoxAndBearingOnGlobe(transform: Transform, p0: LngLatLike, p1: LngLatLike, bearing: number, options?: CameraOptions): ?EasingOptions {
+        const tr = transform.clone();
+        const eOptions = this._extendCameraOptions(options);
+
+        tr.bearing = bearing;
+
+        const coord0 = LngLat.convert(p0);
+        const coord1 = LngLat.convert(p1);
+        const origin = latLngToECEF(
+            (coord0.lat + coord1.lat) * 0.5,
+            (coord0.lng + coord1.lng) * 0.5);
+
+        const zAxis = vec3.normalize([], origin);
+        const xAxis = vec3.normalize([], vec3.cross([], zAxis, [0, 1, 0]));
+        const yAxis = vec3.cross([], xAxis, zAxis);
+
+        const aabbOrientation = [
+            xAxis[0], xAxis[1], xAxis[2], 0,
+            yAxis[0], yAxis[1], yAxis[2], 0,
+            zAxis[0], zAxis[1], zAxis[2], 0,
+            0, 0, 0, 1
+        ];
+        const min = [Infinity, Infinity, Infinity];
+        const max = [-Infinity, -Infinity, -Infinity];
+
+        const ecefCoords = [];
+
+        ecefCoords.push(origin);
+
+        ecefCoords.push(latLngToECEF(coord0.lat, coord0.lng));
+        ecefCoords.push(latLngToECEF(coord1.lat, coord0.lng));
+        ecefCoords.push(latLngToECEF(coord1.lat, coord1.lng));
+        ecefCoords.push(latLngToECEF(coord0.lat, coord1.lng));
+
+        ecefCoords.push(latLngToECEF((coord0.lat + coord1.lat) * 0.5, coord0.lng));
+        ecefCoords.push(latLngToECEF((coord0.lat + coord1.lat) * 0.5, coord1.lng));
+        ecefCoords.push(latLngToECEF(coord0.lat, (coord0.lng + coord1.lng) * 0.5));
+        ecefCoords.push(latLngToECEF(coord1.lat, (coord0.lng + coord1.lng) * 0.5));
+
+        for (const p of ecefCoords) {
+            min[0] = Math.min(min[0], vec3.dot(xAxis, p));
+            min[1] = Math.min(min[1], vec3.dot(yAxis, p));
+            min[2] = Math.min(min[2], vec3.dot(zAxis, p));
+
+            max[0] = Math.max(max[0], vec3.dot(xAxis, p));
+            max[1] = Math.max(max[1], vec3.dot(yAxis, p));
+            max[2] = Math.max(max[2], vec3.dot(zAxis, p));
+        }
+
+        const aabb = new Aabb(min, max);
```

Comment:
You could do `ecefCoords.map(p => [vec3.dot(xAxis, p), vec3.dot(yAxis, p), vec3.dot(zAxis, p)])` and then use `Aabb.fromPoints` to remove some more repetitive code.